prboom2/src/gl_missingtexture.c

```diff
@@ -171,53 +171,46 @@ static void gld_PrepareSectorSpecialEffects(void)
 static void gld_RegisterBleedthroughSector(sector_t* source, sector_t* target, int ceiling)
 {
   int i;
-  int source_idx = -1;
+  int idx = -1;
   assert(source);
   assert(target);
 
-  /* check whether the sector is processed already */
-  for (i = 0; i < numbleedsectors && source_idx == -1; i++)
-    if (bleedsectors[i].source == source && bleedsectors[i].ceiling == ceiling)
-      source_idx = i;
+  /* check whether the target sector is processed already */
+  for (i = 0; i < numbleedsectors && idx == -1; i++)
+    if (bleedsectors[i].target == target &&
+        bleedsectors[i].ceiling == ceiling)
+      idx = i;
 
-  if (source_idx == -1)
+  if (idx == -1)
   {
     /* allocate memory for new sector */
     bleedsectors = (bleedthrough_t*) Z_Realloc(bleedsectors, (numbleedsectors + 1) * sizeof(bleedthrough_t));
     if(!bleedsectors) I_Error("gld_RegisterBleedthroughSector: Out of memory");
     memset(&bleedsectors[numbleedsectors], 0, sizeof(bleedthrough_t));
     numbleedsectors++;
 
-    source_idx = numbleedsectors - 1;
+    idx = numbleedsectors - 1;
   }
 
-  bleedsectors[source_idx].source = source;
-  bleedsectors[source_idx].ceiling = ceiling;
+  bleedsectors[idx].ceiling = ceiling;
+  bleedsectors[idx].target = target;
 
-  /* either register the proposed target since it is first,
-   * or check if the new proposed target is a better option
+  /* either register the proposed source since it is first,
+   * or check if the new proposed source is a better option
    * and register it instead */
-  if (
-    bleedsectors[source_idx].target == NULL ||
-    (
-      bleedsectors[source_idx].target &&
-      (
-        (ceiling && bleedsectors[source_idx].target->ceilingheight > target->ceilingheight) ||
-        bleedsectors[source_idx].target->floorheight < target->floorheight
-      )
-    )
-  )
-  {
-    bleedsectors[source_idx].target = target;
-  }
+  if (bleedsectors[idx].source == NULL ||
```

Comment:
This is equivalent right? I think that'll be easier to parse.
```c
!bleedsectors[idx].source ||
(ceiling && bleedsectors[idx].source->ceilingheight > source->ceilingheight) ||
bleedsectors[idx].source->floorheight < source->floorheight
```
By the way just looking at this...should it be `(!ceiling && bleedsectors[idx].source->floorheight < source->floorheight)` as well, or is it just not a symmetric condition?

Reply:
The original had the same asymmetry, but I think you're right that it's a bug.